projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -1156,29 +1159,34 @@ export class IgxTimePickerComponent implements
      */
     public openDialog(timePicker: IgxTimePickerComponent = this): void {
 
-        if (this.collapsed) {
-            this.collapsed = false;
-
+        if (this.toggleRef.collapsed) {
             let settings;
             if (this.mode === InteractionMode.Dialog) {
                 settings = this.overlaySettings || this._dialogOverlaySettings;
             }
 
             if (this.mode === InteractionMode.DropDown) {
                 settings = this.overlaySettings || this._dropDownOverlaySettings;
-                if (settings.positionStrategy && settings.positionStrategy.settings) {
-                    settings.positionStrategy.settings.target = this.group.element.nativeElement;
+                const posStrategy = settings.positionStrategy;
+
+                if (this.group && posStrategy) {
+                    posStrategy.settings.target = this.group.element.nativeElement;
+                } else if (this.templateDropDownTarget && posStrategy) {
+                    posStrategy.settings.target = this.templateDropDownTarget.nativeElement;
+                } else if (!posStrategy || (posStrategy && !posStrategy.settings.target)) {
+                    throw new Error('There is no target element for the dropdown to attach.' +
+                        'Mark a DOM element with #dropDownTarget ref variable or provide correct overlay positionStrategy.');
                 }
             }
 
             if (this.outlet) {
                 settings.outlet = this.outlet;
             }
 
-            this._overlayId = this.overlayService.attach(this.container, settings);
-            this.overlayService.show(this._overlayId);
+            this.toggleRef.open(settings);
 
         } else if (this.mode === InteractionMode.DropDown) {
+            this.toggleRef.close();
```

Comment:
Calling close() in openDialog() looks strange to me. Probably you should refactor or at least rename this.

Reply:
I would rather keep the name the way it is. We expose **openDialog**  in the context object and changing the name will be a breaking change for the users. I agree that calling **close()** in **openDialog()** does not make much sense but it is necessary for toggling the dropdown when clicking on the icon. I was thinking of using the **toggle()** method but it does not work well when having a retemplated input group.